packages/grid/_modules_/grid/hooks/features/focus/useGridFocus.ts

```diff
@@ -80,6 +90,48 @@ export const useGridFocus = (apiRef: GridApiRef): void => {
     }));
   }, [logger, setGridState]);
 
+  const handleCellMouseUp = React.useCallback(
+    (params: GridCellParams) => {
+      const { cell } = apiRef.current.getState().focus;
+      if (!cell) {
+        return;
+      }
+
+      if (params.id === cell.id && params.field === cell.field) {
+        insideFocusedCell.current = true;
```

Comment:
Did you copy the solution I was proposing for fixing ClickAwayListener with the Select? 😁